app-creator/src/widgets/ui-map/ui-map.ts

```diff
@@ -1,72 +1,229 @@
 import {} from 'googlemaps';
-import { customElement, html, LitElement } from 'lit-element';
+import { customElement, html, LitElement, property, css } from 'lit-element';
+import { nothing } from 'lit-html';
+import { InputType } from '../../redux/types/enums';
+import {
+  AttributeMetaData,
+  DefaultAttributesType,
+  getDefaultAttributes,
+} from '../../redux/types/attributes';
+import { store } from '../../redux/store';
+import { updateWidgetRef, setEditingWidget } from '../../redux/actions';
+
+declare global {
+  interface Window {
+    __initGoogleMap: any;
+  }
+}
+
+window.__initGoogleMap = window.__initGoogleMap || {};
+
+let initCalled: boolean = false;
+const callbackPromise = new Promise((r) => (window.__initGoogleMap = r));
+
+function loadGoogleMaps(apiKey: string) {
+  if (!initCalled) {
+    const script = document.createElement('script');
+    script.src =
+      'https://maps.googleapis.com/maps/api/js?' +
+      (apiKey ? `key=${apiKey}&` : '') +
+      'callback=__initGoogleMap';
+    document.head.appendChild(script);
+    initCalled = true;
+  }
+  return callbackPromise;
+}
```

Comment:
If users add another ui-map to the canvas, would google maps api be reinitialized? Is there a way to reduce the redundancy?

Reply:
After the initial call, initCalled will be set to true, so we won't be re-initializing, we will just return the callback promise.

Reply:
I've updated it to this: 
```
if (this.map == null) {
        this.map = new google.maps.Map(this, this.mapOptions);
      } else {
        this.map.setOptions(this.mapOptions);
      }
```

Reply:
Since we are initializing `this.map` within the instance, next time it might still be undefined when we initialize a new instance. Could we instead include the script in HTML like below:
```
<script src="https://maps.googleapis.com/maps/api/js?key=YOUR_API_KEY&callback=initMap&libraries=&v=weekly" defer></script>
```,
and the `google.maps.Map` namespace should be available across the entire UI.

See https://developers.google.com/maps/documentation/javascript/tutorial#The_Hello_World_of_Google_Maps_v3 for a detailed example.